### PR TITLE
Introduce a validation config for validating operation using rosetta-cli

### PR DESCRIPTION
**Pull request by shrimalmadhur**
Fixes # .

### Motivation
There are lots of validation `rosetta-cli` doesn't perform on operation. 
Few of them include
- validating that deposit and withdraw balances
- issuers might want to validate the number of operation for payment or fee

### Solution
We introduce a validation config file which an asset issuer can specify and provide as an argument while doing `check:data` (more to come on how to use)

TODO:
- Add more tests
- Figure out how to release the package so that it doesn't break existing usage

### CODE REVIEW

File: asserter/asserter.go [hunk 1]
```diff
@@ -147,6 +186,9 @@ func NewClientWithFile(
 		config.AllowedOperationStatuses,
 		config.AllowedErrors,
 		&config.AllowedTimestampStartIndex,
+		&Validations{
+			Enabled: false,
```
Comment (xiaying-peng) [inline asserter/asserter.go:190]: Could you add a comment why we do not need validation config here?

File: asserter/block.go [hunk 2]
```diff
@@ -293,12 +293,31 @@ func (a *Asserter) Operations(
...
+	zero := big.NewInt(0)
 	for i, op := range operations {
 		// Ensure operations are sorted
 		if err := a.Operation(op, int64(i), construction); err != nil {
 			return err
 		}
 
+		if a.validations.Enabled {
+			if op.Type == a.validations.Payment.Name {
+				val, _ := new(big.Int).SetString(op.Amount.Value, 10)
```
Comment (xiaying-peng) [inline asserter/block.go:309]: I think the result boolean matters, which indicate conversion succeed or not.

File: asserter/block.go [hunk 3]
```diff
@@ -293,12 +293,31 @@ func (a *Asserter) Operations(
...
+				paymentTotal.Add(paymentTotal, val)
+				paymentCount++
+			}
+
+			if op.Type == a.validations.Fee.Name {
+				val, _ := new(big.Int).SetString(op.Amount.Value, 10)
```
Comment (xiaying-peng) [inline asserter/block.go:315]: same here, fail if unable to convert.

File: asserter/block_test.go [hunk 4]
```diff
@@ -268,6 +268,134 @@ func TestAccountIdentifier(t *testing.T) {
...
+		}
+	)
+	var tests = map[string]struct {
+		operations         []*types.Operation
+		validationFilePath string
+		asserter           *Asserter
+		construction       bool
+		err                error
+	}{
+		"valid operations based on validation file": {
```
Comment (xiaying-peng) [inline asserter/block_test.go:307]: Add failure cases. Also bad amount cases: make to_string conversion failed.

File: fetcher/fetcher_test.go [hunk 5]
```diff
@@ -189,7 +189,7 @@ func TestInitializeAsserter(t *testing.T) {
 				WithRetryElapsedTime(5*time.Second),
 			)
 
-			networkIdentifier, networkStatus, err := f.InitializeAsserter(ctx, test.network)
+			networkIdentifier, networkStatus, err := f.InitializeAsserter(ctx, test.network, "")
```
Comment (xiaying-peng) [inline fetcher/fetcher_test.go:192]: cannot believe Golang does not support default value..

File: asserter/asserter.go [hunk 6]
```diff
@@ -261,3 +305,20 @@ func (a *Asserter) OperationSuccessful(operation *types.Operation) (bool, error)
 
 	return val, nil
 }
+
+func getValidationConfig(validationFilePath string) (*Validations, error) {
+	validationConfig := &Validations{
+		Enabled: false,
```
Comment (xiaying-peng) [inline asserter/asserter.go:311]: should we enable it by default?

File: asserter/block.go (hunk 2 above)
Comment (shrimalmadhur) [reply asserter/block.go:309]: We are already verifying this with the `Operation` call at line 303. 
https://github.com/coinbase/rosetta-sdk-go/blob/master/asserter/block.go#L53 - so we don't need it here

File: asserter/block.go (hunk 3 above)
Comment (shrimalmadhur) [reply asserter/block.go:315]: We are already verifying this with the `Operation` call at line 303. 
https://github.com/coinbase/rosetta-sdk-go/blob/master/asserter/block.go#L53 - so we don't need it here

File: asserter/asserter.go (hunk 6 above)
Comment (shrimalmadhur) [reply asserter/asserter.go:311]: I think to support backward compatibility, it makes sense to disable it by default.

File: fetcher/fetcher_test.go (hunk 5 above)
Comment (shrimalmadhur) [reply fetcher/fetcher_test.go:192]: 🤷

File: asserter/asserter.go (hunk 1 above)
Comment (shrimalmadhur) [reply asserter/asserter.go:190]: `rosetta-cli` doesn't use this function. I tried to limit this functionality to that so that's why disabled it.

**Review by xiaying-peng — APPROVED** [review]
LGTM!

**Pull request merged**